src/metabase/api/database.clj

```diff
@@ -718,8 +718,8 @@
 (api/defendpoint DELETE "/:id"
   "Delete a `Database`."
   [id]
-  (let [db (api/write-check (Database id))]
-    (api/write-check db)
+  (api/check-superuser)
+  (let [db (Database id)]
```

Comment:
this should be `api/check-404` so we don't attempt to delete a database that does not exist.

Reply:
or `(api/let-404 [database (Database id)]`